src/problems/shinny_sam_sudoku.clj

```diff
@@ -0,0 +1,150 @@
+
+                                  [8 2 6 1 9 5 3 4 7]
+                                  [3 7 4 6 8 2 9 1 5]
+                                  [0 5 0 7 4 3 6 2 8]
+
+                                  [5 1 9 3 2 6 8 7 4]
+                                  [2 4 8 9 5 7 1 3 6]
+                                  [7 6 3 4 1 8 2 5 9]] [0, 0]))
+
+
+
+
+;go through the number 1-9 to put in the current position and produce different childrens of States of the board(haven't implement the legal function yet)
+(for [num [1 2 3 4 5 6 7 8 9]]
+  (->State (add (:board  start-state)
+                (:curr-position  start-state) num numBank) (updatePos start-state)))
+
+
+;divide into 3*3 and return a coll
+(defn grid-1 [board]
+  (for [x[0 1 2] y[0 1 2]]
+    (get (get board x) y)))
+
+(defn grid-2 [board]
+  (for [x[0 1 2] y[3 4 5]]
+    (get (get board x) y)))
+
+(defn grid-3 [board]
+  (for [x[0 1 2] y[6 7 8]]
+    (get (get board x) y)))
+
+(defn grid-4 [board]
+  (for [x[3 4 5] y[0 1 2]]
+    (get (get board x) y)))
+
+(defn grid-5 [board]
+  (for [x[3 4 5] y[3 4 5]]
+    (get (get board x) y)))
+
+(defn grid-6 [board]
+  (for [x[3 4 5] y[6 7 8]]
+    (get (get board x) y)))
+
+(defn grid-7 [board]
+  (for [x[6 7 8] y[0 1 2]]
+    (get (get board x) y)))
+
+(defn grid-8 [board]
+  (for [x[6 7 8] y[3 4 5]]
+    (get (get board x) y)))
+
+(defn grid-9 [board]
+  (for [x[6 7 8] y[6 7 8]]
+    (get (get board x) y)))
+
+(defn board-legal? [state x y]
+
+  (let       [row (row-seq (:board state) x)
+             col (col-seq (:board state) y)]
+          (and (legal? row) (legal? col) (legal? (grid-1(:board state))) (legal? (grid-2(:board state))) (legal? (grid-3(:board state)))
+                                         (legal? (grid-4(:board state))) (legal? (grid-5(:board state))) (legal? (grid-6(:board state)))
+                                         (legal? (grid-7(:board state))) (legal? (grid-8(:board state))) (legal? (grid-9(:board state)))))
+          )
+
+
+;children function
+(defn children [state]
+  (let [x (first (:curr-position state))
+        y (second (:curr-position state))]
+    (if (not= 0 (get (get (:board start-state) x) y)) (->State (:board state) (updatePos state))
+    (for [num [1 2 3 4 5 6 7 8 9]
+          :when (board-legal? (->State (add (:board  state) (:curr-position  state) num numBank) (updatePos state)) x y)
+          ]
+      (->State (add (:board  state)
+                    (:curr-position  state) num numBank) (updatePos state))))))
+
+
+;visual representation of a current sudoku board example's axis (currently operates in [y x])
+
+;    x0 x1 x2 x3 x4 x5 x6 x7 x8
+;y0  [0  1  2  0  0  0  0  0  0]
+;y1  [3  4  5  0  0  0  0  0  0]
+;y2  [6  7  8  0  0  0  0  0  0]
+
+;y3  [0  1  2  0  0  0  0  0  0]
+;y4  [3  4  5  0  0  0  0  0  0]
+;y5  [6  7  8  0  0  0  0  0  0]
+
+;y6  [0  1  2  0  0  0  0  0  0]
+;y7  [3  4  5  0  0  0  0  0  0]
+;y8  [6  7  8  0  0  0  0  0  0]
+
```

Comment:
Notice all these blank lines at the end; you probably want to clean those up. You might also think about whether you want the grid before; that might be useful to keep, or you might consider it historical clutter.